lib/ansible/modules/uri.py

```diff
@@ -726,16 +688,50 @@ def main():
 
     # Make the request
     start = datetime.datetime.utcnow()
-    resp, content, dest = uri(module, url, dest, body, body_format, method,
-                              dict_headers, socket_timeout, ca_path, unredirected_headers)
-    resp['elapsed'] = (datetime.datetime.utcnow() - start).seconds
+    r, info = uri(module, url, dest, body, body_format, method,
+                  dict_headers, socket_timeout, ca_path, unredirected_headers)
+
+    elapsed = (datetime.datetime.utcnow() - start).seconds
+
+    if r and dest is not None and os.path.isdir(dest):
+        filename = get_response_filename(r) or 'index.html'
+        dest = os.path.join(dest, filename)
+
+    if r:
+        content_type, main_type, sub_type, content_encoding = parse_content_type(r)
+    else:
+        content_type = 'application/octet-stream'
+        main_type = 'aplication'
+        sub_type = 'octet-stream'
+        content_encoding = 'utf-8'
+
+    maybe_json = content_type and any(candidate in sub_type for candidate in JSON_CANDIDATES)
+    maybe_output = maybe_json or return_content or info['status'] not in status_code
+
+    if maybe_output:
+        try:
+            if PY3 and r.closed:
+                raise TypeError
+            content = r.read()
+        except (AttributeError, TypeError):
+            # there was no content, but the error read()
+            # may have been stored in the info as 'body'
+            content = info.pop('body', b'')
+    elif r:
```

Comment:
little nit - there's no `else` to ensure content is always defined by the time it's used

Reply:
I just made this a draft to prevent the bot from merging until I can look at this properly.  I obviously don't have any tests that trigger this, and I wonder if it's even possible to fall past this.

Reply:
Ok, I just added the `else: content = None`

Reply:
Maybe if an exception other than an HTTPError occurs in fetch_url (since in that case `r = e`)? But I'm not sure if it's really an issue.